### PR TITLE
[Framework] AIP-45: Allow soft burning objects

### PR DESCRIPTION
**Pull request by movekevin**
### Description
Context: https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-45.md

### Test Plan
Unit tests. Manual tests in devnet/testnet/mainnet later

### CODE REVIEW

**Review by davidiw — APPROVED** [review]
you're such a perfectionist.....

**Comment by bowenyang007** [comment]
What are we doing for secondary stores?

**Review by bowenyang007 — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/object.move [hunk 1]
```diff
@@ -84,6 +86,9 @@ module aptos_framework::object {
     /// derivation to produce an object address.
     const OBJECT_FROM_SEED_ADDRESS_SCHEME: u8 = 0xFE;
 
+    /// Address where unwanted objects can be forcefully transferred to.
+    const BURN_ADDRESS: address = @0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
```
Comment (lightmark) [inline aptos-move/framework/aptos-framework/sources/object.move:90]: I wonder how we choose between 0x0 and 0xffffffffff.... 0 is not that error prone to me.

**Review by lightmark — COMMENTED** [review]
do we want a short security review for this?

File: aptos-move/framework/aptos-framework/sources/primary_fungible_store.move [hunk 2]
```diff
@@ -126,6 +126,8 @@ module aptos_framework::primary_fungible_store {
     /// Withdraw `amount` of fungible asset from the given account's primary store.
     public fun withdraw<T: key>(owner: &signer, metadata: Object<T>, amount: u64): FungibleAsset {
         let store = primary_store(signer::address_of(owner), metadata);
+        // Check if the store object has been burnt or not. If so, unburn it first.
+        may_be_unburn(owner, store);
```
Comment (lightmark) [inline aptos-move/framework/aptos-framework/sources/primary_fungible_store.move:130]: Why is this necessary? it seems fine to error out in this case.

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/primary_fungible_store.move:130]: To make it seem more seamless to transition between burn and unburn

File: aptos-move/framework/aptos-framework/sources/object.move (hunk 1 above)
Comment (xbtmatt) [reply aptos-move/framework/aptos-framework/sources/object.move:90]: Also, does it matter if this is a reserved framework address or not?
